oak/server/labels.cc

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2019 The Project Oak Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "oak/server/labels.h"
+
+#include <algorithm>
+
+namespace oak {
+
+void OakLabels::Merge(const oak::OakLabels& other) {
```

Comment:
I would be tempted to leave out the merge operation and the "floating" labels. Basically, given the current state of the labels on a node and on an incoming message, determine whether the message can be read by the node at all, without changing any labels. Then separately we should discuss how nodes change their own labels (e.g. using capabilities). But fine if you want to keep this around for now too.